pkg/controller/egress/controller.go

```diff
@@ -311,6 +315,37 @@ func (c *EgressController) setIPAllocation(egressName string, ip net.IP, poolNam
 	}
 }
 
+func (c *EgressController) updateExternalIPPoolStatus(poolName string, total, used int) error {
```

Comment:
This method can be called concurrently in pool's eventHandler and egress's workers. So it seems that a race condition can make they override each other and the final value in API may be wrong. Maybe creating multiple Egresses can reproduce it.
If this is true, I would suggest have dedicated workers to sync pool's status, other places only enqueue a pool that needs to sync.

Reply:
Good consideration. Will reproduce and fix it.